[Guo Yulong] iP

gyulong1 opened the pull request:
# DukePro
> "“Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes)

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/admin/ip-w4.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

gyulong1 committed:
Added Level-1

gyulong1 committed:
Added Level-2

gyulong1 committed:
Added Level-3

gyulong1 committed:
Added Level-4

gyulong1 committed:
Added A-TextUiTesting

gyulong1 committed:
Added Level-5

gyulong1 committed:
Added A-Collections

gyulong1 committed:
Add Level-7

gyulong1 committed:
Add Level-8

gyulong1 committed:
Merge remote-tracking branch 'origin/branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

gyulong1 committed:
Add A-MoreOOP

gyulong1 committed:
Add A-Packages

gyulong1 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

gyulong1 committed:
Add A-Gradle

gyulong1 committed:
Add A-JUnit

gyulong1 committed:
Add A-JavaDoc

gyulong1 committed:
Add A-CodingStandard

gyulong1 committed:
Add Level-9

gyulong1 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/command/CommandDelete.java
#	src/main/java/duke/command/CommandTask.java

gyulong1 committed:
Merge branch 'branch-Level-9'

bernicetoh commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,43 @@
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
```
Perhaps you can include JavaDoc comments for each class?

bernicetoh commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,22 @@
+
+        assertEquals(true, command.isExit());
+    }
+}
```
Looks good.

bernicetoh reviewed (commented):
Overall, I found your code neat and have followed the coding standard. I noticed that you have not added JavaDoc comments for your classes. Consider including them in.

gyulong1 committed:
Add Checkstyle

rockman007372 commented on src/main/java/duke/Parser.java:
```diff
@@ -72,7 +76,9 @@ public static Command parse(String fullCommand) {
                     int indexTo = fullCommand.indexOf("/to");
                     if (indexFrom != -1 && indexTo != -1) {
-                        t = new Event(fullCommand.substring(6, indexFrom), fullCommand.substring(indexFrom + 6, indexTo - 1), fullCommand.substring(indexTo + 4));
+                        t = new Event(fullCommand.substring(6, indexFrom),
```
Neat indexing that follows coding standard. Well done!

rockman007372 commented on src/test/java/duke/ParserTest.java:
```diff
@@ -1,22 +1,23 @@
 public class ParserTest {
     @Test
-    public void dummyTest(){
+    public void dummyTest() {
```
I think using another name for test method would be more appropriate. Perhaps using the test name convention `whatIsBeingTested_descriptionOfTestInputs_expectedOutcome`

rockman007372 commented on src/main/java/duke/task/Task.java:
```diff
@@ -25,7 +28,7 @@ public String getDesc() {
      * @param isDone final status of the task.
      */
-    public void setStatus(Boolean isDone){
+    public void setStatus(Boolean isDone) {
```
I like how you combine mark and unmark method into a single setStatus method!

rockman007372 commented on src/main/java/duke/Parser.java:
```diff
@@ -72,7 +76,9 @@ public static Command parse(String fullCommand) {
                     int indexFrom = fullCommand.indexOf("/from");
                     int indexTo = fullCommand.indexOf("/to");
                     if (indexFrom != -1 && indexTo != -1) {
-                        t = new Event(fullCommand.substring(6, indexFrom), fullCommand.substring(indexFrom + 6, indexTo - 1), fullCommand.substring(indexTo + 4));
```
Good parsing practices but I also recommend using method `trim()` to remove the leading and trailing whitespace in the task description!

rockman007372 reviewed (commented):
Overall neat and readable code that follows Java coding convention well. My only suggestions are to consider trailing whitespace during parsing and to rename the test methods to be more in line with convention! Great job Yulong!

jengoc415 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
```
I like how you stored the information!

jengoc415 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,96 @@
+    public static Command parse(String fullCommand) {
+
+        if (fullCommand.equalsIgnoreCase("bye")) {
+            return new CommandBye(fullCommand);
```
Might want to consider using a switch case to make code more readable

jengoc415 commented on src/main/java/duke/command/CommandDelete.java:
```diff
@@ -0,0 +1,39 @@
+        System.out.println("Now you have " + len + " in the list.");
+
+        storage.writeArray(taskList);
+    }
```
Neat code even printing complex set of text

jengoc415 commented on src/main/java/duke/command/CommandFind.java:
```diff
@@ -0,0 +1,38 @@
+        System.out.println("Here are the matching tasks in your list:\n");
+        for (int i = 0; i < taskList.getLength(); i++) {
+            if (taskList.getTask(i).getDesc().contains(word)) {
+                System.out.printf("%d.%s\n", i + 1, taskList.getTask(i).toString());
```
Might be more readable to split these functions up instead of doing it all in one line

jengoc415 reviewed (commented):
Great job on the code overall!

gyulong1 committed:
Add Level-10

gyulong1 committed:
Add A-Assertions

Added assertions to check that input line should be non-empty as this can cause exceptions.

Also checked if the requested index was within the length of Tasklist before marking.

gyulong1 committed:
Add A-CodeQuality

Refactored code to make it more readable using switch statements and more intermediate variables.

gyulong1 committed:
Merge pull request #1 from gyulong1/branch-A-Assertions

Add A-Assertions

gyulong1 committed:
Changed import messages

gyulong1 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/task/Task.java

gyulong1 committed:
Merge pull request #2 from gyulong1/branch-A-CodeQuality

Add A-CodeQuality

gyulong1 committed:
Undo Command: add undo feature

Ability to undo to add more flexiblity for the user.

Creating a temporary store to the arraylist for the previous command. Introduce inverse commands which undo the effect of the previous command.

To reduce adding more classes and utilise the existing abstraction to achieve the feature.

gyulong1 committed:
Revert "Undo Command: add undo feature"

This reverts commit 8501c167c7612acfd56795ce41255bde42219fa7.

gyulong1 committed:
Revert "Revert "Undo Command: add undo feature""

This reverts commit d294575e944bbbc840b25aa9280772da3358baae.

gyulong1 committed:
Change user picture to one that has a transparent background.

Fixed bugs with undo feature.

gyulong1 committed:
Credited checkstyle

Change import statements

gyulong1 committed:
Add name in title

gyulong1 committed:
Add user guide draft

Included examples of features and screenshot of start UI.

gyulong1 committed:
Change exit command to close the window.

Update import order to align with the guidelines.

gyulong1 committed:
Update naming of variables and header to adhere to coding standard.
